bin/init.sh

```diff
@@ -42,55 +44,81 @@ if [ ${ROOT} != "${GO_TOP:-$HOME/go}/src/istio.io/istio" ]; then
+# docker images created by Dockerfile.proxy and Dockerfile.proxy_debug.
 
-    DOWNLOAD_COMMAND=""
+# Gets the download command supported by the system (currently either curl or wget)
+DOWNLOAD_COMMAND=""
+set_download_command () {
+    # Try curl.
     if command -v curl > /dev/null; then
-       if curl --version | grep Protocols  | grep https > /dev/null; then
-	   DOWNLOAD_COMMAND='curl -fLSs'
-       else
-           echo curl does not support https, will try wget for downloading files.
-       fi
+        if curl --version | grep Protocols  | grep https > /dev/null; then
+	       DOWNLOAD_COMMAND='curl -fLSs'
+	       return
+        fi
+        echo curl does not support https, will try wget for downloading files.
     else
-       echo curl is not installed, will try wget for downloading files.
+        echo curl is not installed, will try wget for downloading files.
     fi
 
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        if command -v wget > /dev/null; then
-	    DOWNLOAD_COMMAND='wget -qO -'
-        else
-            echo wget is not installed.
-        fi
+    # Try wget.
+    if command -v wget > /dev/null; then
+        DOWNLOAD_COMMAND='wget -qO -'
+        return
     fi
+    echo wget is not installed.
 
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        echo Error: curl is not installed or does not support https, wget is not installed. \
-             Cannot download envoy. Please install wget or add support of https to curl.
-        exit 1
-    fi
+    echo Error: curl is not installed or does not support https, wget is not installed. \
+         Cannot download envoy. Please install wget or add support of https to curl.
+    exit 1
+}
+
+# Keep the default URLs up-to-date with the latest push from istio/proxy.
+export ISTIO_ENVOY_DIR=${GO_TOP}/out
+export ISTIO_ENVOY_VERSION=${ISTIO_ENVOY_VERSION:-54077a0c8486b6db7b2fc5d10619b6813bb40767}
+export ISTIO_ENVOY_DEBUG_URL=${ISTIO_ENVOY_DEBUG_URL:-https://storage.googleapis.com/istio-build/proxy/envoy-debug-${ISTIO_ENVOY_VERSION}.tar.gz}
+export ISTIO_ENVOY_RELEASE_URL=${ISTIO_ENVOY_RELEASE_URL:-https://storage.googleapis.com/istio-build/proxy/envoy-alpha-${ISTIO_ENVOY_VERSION}.tar.gz}
+
+# Variables for the extracted debug/release artifacts.
+export ISTIO_ENVOY_DEBUG=envoy-debug-$ISTIO_ENVOY_VERSION
+export ISTIO_ENVOY_DEBUG_PATH="$ISTIO_ENVOY_DIR/$ISTIO_ENVOY_DEBUG"
+export ISTIO_ENVOY_RELEASE=envoy-$ISTIO_ENVOY_VERSION
+export ISTIO_ENVOY_RELEASE_PATH="$ISTIO_ENVOY_DIR/$ISTIO_ENVOY_RELEASE"
+
+# Save envoy in $ISTIO_ENVOY_DIR
+if [ ! -f "$ISTIO_ENVOY_DIR/$ISTIO_ENVOY_DEBUG" ] || [ ! -f "$ISTIO_ENVOY_DIR/$ISTIO_ENVOY_RELEASE" ] ; then
+    mkdir -p $ISTIO_ENVOY_DIR
+    pushd $ISTIO_ENVOY_DIR
+
+    # Set the value of DOWNLOAD_COMMAND (either curl or wget)
+    set_download_command
+
+    # Download debug envoy binary.
+    echo "Downloading envoy debug artifact: ${DOWNLOAD_COMMAND} ${ISTIO_ENVOY_DEBUG_URL}"
+    time ${DOWNLOAD_COMMAND} ${ISTIO_ENVOY_DEBUG_URL} | tar xz
```

Comment:
I think tar can be told to extract the one file of interest (probably as "./usr/local/bin/envoy").

Reply:
There is only one file anyway.